src/H5FDint.c

```diff
@@ -1170,7 +1170,7 @@ H5FD_read_selection(H5FD_t *file, H5FD_mem_t type, uint32_t count, H5S_t **mem_s
 
             if ((file_space_ids[num_spaces] = H5I_register(H5I_DATASPACE, file_spaces[num_spaces], TRUE)) <
                 0) {
-                if (H5I_dec_app_ref(mem_space_ids[num_spaces]) < 0)
+                if (NULL == H5I_remove(mem_space_ids[num_spaces]))
                     HDONE_ERROR(H5E_VFL, H5E_CANTDEC, FAIL, "problem freeing id")
```

Comment:
Does error code/message need to be changed accordingly?  Same below.

Reply:
I think it's still valid - the ID itself is still being freed.

Reply:
Maybe H5E_CANTDEC should be changed